### DISCUSSION TITLE
Fire a webhook when a file has been updated/removed

### DISCUSSION THREAD

**Discussion by damien-mcmahon** [directus/directus#21569] [Feature Requests]
### Summary

We want to invalidate caching of images that have been updated. Also when an image is removed, we want to clear up our optimised bucket which contains images that have been optimised with Sharp

### Basic Example

In the UI, it's possible to select a file instead of a model from the list in the Webhooks UI

### Motivation

We would like a way of reliably controlling the assets we cache and serve to the user. We also want to reduce usage in our S3 bucket of images no longer in use in the CMS.

### Detailed Design

Use the existing webhook design but enable it for use on files. It wouldn't need to have any filtering capabilities, just being notified when the resource has been updated/removed is enough.

### Must Have:
- A webhook that gets fired when a resource is updated
- A webhook that gets fired when a resouce is deleted

### Should Have:
- File data sent in the webhook so that identifying which resource in S3 to be removed or which Cloudfront invalidation

### Could Have:
- The ability to filter on file metadata

### Won't Have:
- 

### Drawbacks

It might fire a great deal if bulk uploading/removing of files

### Alternatives

Some kind of constant file monitoring via the REST api?

### Adoption Strategy

In the Webhooks section of Settings, they'll be able to select to fire webhooks on a file remove/update event

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
